### ISSUE TITLE
`delete_tmp_upload` causes Workbench to crash

### ISSUE THREAD

**Issue by rosiel** [mjordan/islandora_workbench#597]
As discussed in [Slack](https://islandora.slack.com/archives/C01TLCB0EEQ/p1683121786286679):

Workbench crashes with a colourful traceback when used with `delete_tmp_upload: true`.
Note, the directory that it's complaining about being missing is `[working dir]/944` when the directory containing the file is at `[working dir]/nid_944`.

It also may(?) be worth noting that I'm not using the `file` column, as none of my media are 'Original File'. Their media use terms are encoded in the config file. Only 'PDF', the third column, is a remote file that needs downloading (the others are local files). 

Error message, config file, and csv sample below:

## Error message

```
"Add media" task started using config file local/config.yml.
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /Users/rlefaive/Git/islandora_workbench/workbench:1382 in <module>                               │
│                                                                                                  │
│   1379 │   if config['task'] == 'delete':                                                        │
│   1380 │   │   delete()                                                                          │
│   1381 │   if config['task'] == 'add_media':                                                     │
│ ❱ 1382 │   │   add_media()                                                                       │
│   1383 │   if config['task'] == 'delete_media':                                                  │
│   1384 │   │   delete_media()                                                                    │
│   1385 │   if config['task'] == 'delete_media_by_node':                                          │
│                                                                                                  │
│ /Users/rlefaive/Git/islandora_workbench/workbench:655 in add_media                               │
│                                                                                                  │
│    652 │   │   │   │   │   │   │   │   │   │   print('- No media for ' + node_uri + ' created s  │
│    653 │   │   │   │   │   │   │   │   │   logging.warning("No media for %s created since its '  │
│    654 │   │   │   │   │   │   │   │   │   continue                                              │
│ ❱  655 │   │   │   │   │   │   │   │   media_response_status_code = create_media(config, row[ad  │
│    656 │   │   │   │   │   │   │   │   if media_response_status_code in allowed_media_response_  │
│    657 │   │   │   │   │   │   │   │   │   if config['progress_bar'] is False:                   │
│    658 │   │   │   │   │   │   │   │   │   │   print("Media for " + row[additional_file_field]   │
│                                                                                                  │
│ /Users/rlefaive/Git/islandora_workbench/workbench_utils.py:2886 in create_media                  │
│                                                                                                  │
│   2883 │   │   # file_result must be an integer.                                                 │
│   2884 │   │   file_result = -1                                                                  │
│   2885 │   else:                                                                                 │
│ ❱ 2886 │   │   file_result = create_file(config, filename, file_fieldname, csv_row, node_id)     │
│   2887 │                                                                                         │
│   2888 │   if filename.startswith('http'):                                                       │
│   2889 │   │   if file_result is False:                                                          │
│                                                                                                  │
│ /Users/rlefaive/Git/islandora_workbench/workbench_utils.py:2829 in create_file                   │
│                                                                                                  │
│   2826 │   │   │   │   containing_folder = os.path.join(config['temp_dir'], re.sub('[^A-Za-z0-9  │
│   2827 │   │   │   │   try:                                                                      │
│   2828 │   │   │   │   │   # E.g., on Windows, "[WinError 32] The process cannot access the fil  │
│ ❱ 2829 │   │   │   │   │   shutil.rmtree(containing_folder)                                      │
│   2830 │   │   │   │   except PermissionError as e:                                              │
│   2831 │   │   │   │   │   logging.error(e)                                                      │
│   2832                                                                                           │
│                                                                                                  │
│ /usr/local/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/sh │
│ util.py:722 in rmtree                                                                            │
│                                                                                                  │
│    719 │   │   try:                                                                              │
│    720 │   │   │   orig_st = os.lstat(path, dir_fd=dir_fd)                                       │
│    721 │   │   except Exception:                                                                 │
│ ❱  722 │   │   │   onerror(os.lstat, path, sys.exc_info())                                       │
│    723 │   │   │   return                                                                        │
│    724 │   │   try:                                                                              │
│    725 │   │   │   fd = os.open(path, os.O_RDONLY, dir_fd=dir_fd)                                │
│                                                                                                  │
│ /usr/local/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/sh │
│ util.py:720 in rmtree                                                                            │
│                                                                                                  │
│    717 │   │   # Note: To guard against symlink races, we use the standard                       │
│    718 │   │   # lstat()/open()/fstat() trick.                                                   │
│    719 │   │   try:                                                                              │
│ ❱  720 │   │   │   orig_st = os.lstat(path, dir_fd=dir_fd)                                       │
│    721 │   │   except Exception:                                                                 │
│    722 │   │   │   onerror(os.lstat, path, sys.exc_info())                                       │
│    723 │   │   │   return                                                                        │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
FileNotFoundError: [Errno 2] No such file or directory: '/Users/rlefaive/Documents/Projects/2023-ilives-migration/02a-ingest-special-datastreams/944'
```

## Config.yml:
```
task: add_media
host: "http://localhost:8000"
username: admin
password: islandora
input_dir: '/Users/rlefaive/Documents/Projects/2023-ilives-migration/02a-ingest-special-datastreams'
input_csv: '999-test.csv'
output_csv: output.csv
output_csv_include_input_csv: false
allow_missing_files: true
use_node_title_for_media_title: true
delete_tmp_upload: true
additional_files:
  - PDF: 37
  - MODS: 34
  - permission: 38
```

## CSV Sample
```
node_id,file,PDF,MODS,permission
944,,http://islandlives.ca/islandora/object/ilives:112483/datastream/PDF/download,MODS/ilives:112483-MODS.xml,permission/ilives:112483-permission.pdf
647,,http://islandlives.ca/islandora/object/ilives:1161945/datastream/PDF/download,MODS/ilives:1161945-MODS.xml,
755,,http://islandlives.ca/islandora/object/ilives:1161946/datastream/PDF/download,MODS/ilives:1161946-MODS.xml,
```

**Comment by mjordan** [comment]
if you add `print(file_path)` right after https://github.com/mjordan/islandora_workbench/blob/main/workbench_utils.py#L2762, what is the output?

**Comment by mjordan** [comment]
Good catch! Can you check out the issue_597 branch and test?

**Comment by rosiel** [comment]
Success! It works!

**Comment by mjordan** [comment]
Thanks, I'm in a meeting now but will merge in a bit.

**Comment by mjordan** [comment]
Resolved with commit 23d2941455a71f4dd2156fc5ad71b00840e4e0d5. Thanks @rosiel for finding this bug!
